File: ruby/rubymonk/non_duplicated_values/spec/non_duplicated_values_spec.rb
require 'spec_helper'
require_relative '../lib/non_duplicated_values'

describe '#non_duplicated_values' do 
  it 'does something' do 
    expect(non_duplicated_values(['ello'])).to_not eq(nil)
  end

  it 'handles a simple case' do 
    expect(non_duplicated_values(['ello'])).to eq(['ello'])
  end

  it 'handles a case where counting is required' do 
    expect(non_duplicated_values(['ello', 'ello', 'foo'])).to eq(['foo'])
  end
end

File: javascript/non_duplicated_values/spec/non_duplicated_values_spec.js
// write a function that takes in an array
// & returns an array of values that are contained only once

var nonDuplicatedValues = require('../lib/non_duplicated_values.js');

describe('nonDuplicatedValues', function() {
    it('is a function', function () {
        expect(typeof nonDuplicatedValues).toBe('function');
    });

    it('handles an empty array', function () {
        expect(nonDuplicatedValues([])).toEqual([]);
    });

    it('handles an array with a single element', function () {
        expect(nonDuplicatedValues(['ello'])).toEqual(['ello']);
    });

    it('handles a simple case', function () {
        expect(nonDuplicatedValues(['foo', 'foo', 'ello'])).toEqual(['ello']);
    });
});